hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -3146,23 +3148,30 @@ INodeFile checkLease(INodesInPath iip, String holder, long fileId)
   boolean completeFile(final String src, String holder,
                        ExtendedBlock last, long fileId)
     throws IOException {
+    final String operationName = CMD_COMPLETE_FILE;
     boolean success = false;
+    FileStatus stat = null;
     checkOperation(OperationCategory.WRITE);
     final FSPermissionChecker pc = getPermissionChecker();
     FSPermissionChecker.setOperationType(null);
     writeLock();
     try {
       checkOperation(OperationCategory.WRITE);
       checkNameNodeSafeMode("Cannot complete file " + src);
-      success = FSDirWriteFileOp.completeFile(this, pc, src, holder, last,
+      INodesInPath iip = dir.resolvePath(pc, src, fileId);
+      success = FSDirWriteFileOp.completeFile(this, iip, src, holder, last,
                                               fileId);
+      if (success) {
+        stat = dir.getAuditFileInfo(iip);
+      }
     } finally {
-      writeUnlock("completeFile");
+      writeUnlock(operationName);
```

Comment:
why change this?

Reply:
Sorry. I should change this to "complete" instead of "close". I change this because the audit log cmd names usually mimic the client api names rather than the rpc method name. For instance, rpc method "startFile" is audit logged as cmd "create".